python/cudf/cudf/core/index.py

```diff
@@ -2820,12 +2820,14 @@ def interval_range(
     elif periods and not freq:
         # if statement for mypy to pass
         if end is not None and start is not None:
-            freq_step = ((end) - start) / periods
-            freq_step = cudf.Scalar(freq_step, dtype="float64").device_value
-            start = cudf.Scalar(start, dtype="float64").device_value
+            # determine if periods are float or integer
+            quotient, remainder = divmod((end - start), periods)
+            if remainder:
+                freq_step = cudf.Scalar((end - start) / periods).device_value
+            else:
+                freq_step = cudf.Scalar(quotient).device_value
+            start = cudf.Scalar(start, dtype=freq_step.dtype).device_value
```

Comment:
What's the behavior of this if `start` is a `float` and `freq_step.dtype` is `int64`? cc @brandon-b-miller

Reply:
Sequence throws an error if `start` and `freq_step` have different dtypes, which is why I decided to use the dtype of `freq_step`. This is the error I get otherwise.

```
RuntimeError: cuDF failure at: ../../../../src/filling/sequence.cu:143: init and step must be of the same type.
```

From the tests I've run it also looks like if  the`start` input is a `float` then `freq_step.dtype` will also always be a `float` because of the `divmod` operation. I'm adding in a few more tests in case I'm missing something

Reply:
Finding that this doesn't give the right result all the time. Changing back to what you suggested we do initially 😅